pkg/pipeline/encode/encode_s3_test.go

```diff
@@ -98,19 +107,85 @@ func Test_EncodeS3(t *testing.T) {
 
 	<-syncChan
 
-	// confirm object names, bucket name
-	// confirm that object created has batchSize=3 entries
 	writer := encodeS3.s3Writer
 	fakeWriter := writer.(*fakeS3Writer)
 	fakeWriter.mutex.Lock()
 	object0 := fakeWriter.objects[0]
+	objectName0 := fakeWriter.objectNames[0]
+
+	// confirm object header fields
 	require.Contains(t, object0, "version")
 	require.Contains(t, object0, "capture_start_time")
 	require.Contains(t, object0, "capture_end_time")
 	require.Contains(t, object0, "number_of_flow_logs")
+
+	// confirm that object created has batchSize=3 entries
 	require.Equal(t, 3, object0["number_of_flow_logs"])
+
+	// confirm object names, bucket name
 	require.Equal(t, "bucket1", fakeWriter.bucketNames[0])
+	expectedSubstring := "account1"
+	require.True(t, strings.Contains(objectName0, expectedSubstring))
+	expectedSubstring = "year"
+	require.True(t, strings.Contains(objectName0, expectedSubstring))
+	expectedSubstring = "month="
+	require.True(t, strings.Contains(objectName0, expectedSubstring))
+	expectedSubstring = "day="
+	require.True(t, strings.Contains(objectName0, expectedSubstring))
+	expectedSubstring = "hour="
+	require.True(t, strings.Contains(objectName0, expectedSubstring))
+	expectedSubstring = "stream-id="
+	require.True(t, strings.Contains(objectName0, expectedSubstring))
+	expectedSubstring = "00000000"
+	require.True(t, strings.Contains(objectName0, expectedSubstring))
+	expectedSubstring = "00000001"
+	require.True(t, strings.Contains(fakeWriter.objectNames[1], expectedSubstring))
+	expectedSubstring = "00000002"
+	require.True(t, strings.Contains(fakeWriter.objectNames[2], expectedSubstring))
```

Comment:
Using `require.Contains()` is preferred over `require.True()`.
The advantage of  `require.Contains()` is that when the condition is not met, the error message is more verbose.

Reply:
updated accordingly